ios/brave-ios/Sources/Brave/Frontend/Browser/Tabs/TabTray/TabTrayController.swift

```diff
@@ -397,27 +400,42 @@ class TabTrayController: AuthenticationController {
       $0.isLayoutMarginsRelativeArrangement = true
     }
 
-    contentStackView.addArrangedSubview(tabTypeSelector)
-    contentStackView.setCustomSpacing(UX.segmentedControlTopInset, after: tabTypeSelector)
+    tabTypeSelectorContainerView.addSubview(tabTypeSelector)
+    contentStackView.addArrangedSubview(tabTypeSelectorContainerView)
+    contentStackView.setCustomSpacing(
+      UX.segmentedControlTopInset,
+      after: tabTypeSelectorContainerView
+    )
     contentStackView.addArrangedSubview(tabContentView)
 
     containerView.addSubview(contentStackView)
 
     if FeatureList.kBraveShredFeature.enabled {
-      containerView.addSubview(shredButton)
-
-      shredButton.snp.makeConstraints { make in
-        make.trailing.equalTo(containerView.readableContentGuide)
-        make.centerY.equalTo(tabTypeSelector)
+      tabTypeSelectorContainerView.addSubview(shredButton)
+
+      shredButton.snp.makeConstraints {
+        $0.leading.greaterThanOrEqualTo(tabTypeSelector.snp.trailing)
+        $0.trailing.equalTo(tabTypeSelectorContainerView.snp.trailing)
+        $0.centerY.equalTo(tabTypeSelectorContainerView)
+        $0.top.greaterThanOrEqualTo(tabTypeSelectorContainerView.snp.top)
+        $0.bottom.lessThanOrEqualTo(tabTypeSelectorContainerView.snp.bottom)
       }
     }
 
     contentStackView.snp.makeConstraints {
       $0.edges.equalTo(containerView.safeAreaLayoutGuide)
     }
 
+    tabTypeSelectorContainerView.snp.makeConstraints {
+      $0.width.equalTo(containerView.layoutMarginsGuide.snp.width)
+    }
+
     tabTypeSelector.snp.makeConstraints {
       $0.width.equalTo(contentStackView.snp.width).dividedBy(2)
+      $0.centerX.equalTo(tabTypeSelectorContainerView.snp.centerX)
+      $0.centerY.equalTo(tabTypeSelectorContainerView.snp.centerY)
+      $0.top.equalTo(tabTypeSelectorContainerView.snp.top)
+      $0.bottom.equalTo(tabTypeSelectorContainerView.snp.bottom)
```

Comment:
I think you can just do `$0.center.top.bottom.equalTo(tabTypeSelectorContainerView)` just fyi

Reply:
[310778b](https://github.com/brave/brave-core/pull/25909/commits/310778bff5036411b8703d4e0c006062eb28a2c1)